[WIP] rename QupZilla to falkon, update to 3.0.1

chrstphrchvz opened the pull request:
#### Description

<!-- Note: it is best make pull requests from a branch rather than from master -->
- Added category KDE as primary category
- Falkon has moved to using CMake, and requires KDE ECM (extra-cmake-modules). However the version of ECM in MacPorts (5.27) is no longer available, so I have updated it (and ECM-devel) to 5.47.0. I have **not** yet tested the remaining ports which depend on ECM.
- Falkon requires ki18n, which was proposed as part of https://trac.macports.org/ticket/49925 but that appears to have stalled. I have added a separate ~~`ki18n`~~ `kf5-ki18n` port in the meantime, if that is appropriate. ~~**Q:** Should this be instead named e.g. `kde-ki18n` or `kf5-ki18n` instead?~~
- Certain optional features are disabled (password management and Python extension support), either because the required libraries aren't in MacPorts yet (kf5 KWallet, PySide2 for Python 3.x), or might be better off enabled using a variant (gnome-keyring).

I'm currently stuck at figuring out how to properly run CMake for Falkon. It appears to be looking one level too deep for `src/main/Info.plist`:
```
CMake Error: Target falkon Info.plist template "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_falkon/falkon/work/falkon-3.0.1/src/main/main/Info.plist" could not be found.
```

For reference, this is how Falkon's [readme](https://github.com/KDE/falkon/blob/bea5bb05d95f8652f1eaef7755609cbc7ff14399/README.md#L5-L13) suggests running CMake:

>== Building ==
>
>Falkon uses CMake, to build it run:
>
>```
>mkdir build && cd build
>cmake ..
>make
>```

[Full main.log for Falkon](https://gist.github.com/chrstphrchvz/0b77c3c8588cc8821993278a88afc0af)

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.13.5 17F77
Xcode 9.4 9F1027a

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?

**Edit:** I didn't see that [@yan12125 beat me to updating ECM by a few hours](https://github.com/macports/macports-ports/pull/2016).

- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?

No tests specified for ECM ~~or ki18n~~.

<span>
<details>
<summary><b>Edit—</b>Tests results for ki18n:</summary>
<pre>
--->  Testing kf5-ki18n
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build"&& /usr/bin/make test
Running tests...
/opt/local/bin/ctest --force-new-ctest-process
Test project /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build
    Start 1: ki18n-declarativetest
1/5 Test #1: ki18n-declarativetest ............   Passed    0.10 sec
    Start 2: ki18n-klocalizedstringtest
2/5 Test #2: ki18n-klocalizedstringtest .......   Passed    0.08 sec
    Start 3: ki18n-ktranscripttest
3/5 Test #3: ki18n-ktranscripttest ............***Failed    0.06 sec
    Start 4: ki18n-ktranscriptcleantest
4/5 Test #4: ki18n-ktranscriptcleantest .......   Passed    0.06 sec
    Start 5: ki18n_install
5/5 Test #5: ki18n_install ....................   Passed    5.55 sec
&nbsp;
80% tests passed, 1 tests failed out of 5
&nbsp;
Total Test time (real) =   5.86 sec
&nbsp;
The following tests FAILED:
          3 - ki18n-ktranscripttest (Failed)
Errors while running CTest
make: *** [test] Error 8
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build" && /usr/bin/make test
Exit code: 2
</pre>
</details>
<br />
</span>

- [x] tried a full install with `sudo port -vst install`?

Fails, see above

- [ ] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

RJVB commented on kde/ki18n/Portfile:
```diff
@@ -0,0 +1,76 @@
+PortGroup           qt5     1.0
+PortGroup           cmake   1.1
+
+name                ki18n
```
kf5-ki18n please

RJVB commented on kde/ki18n/Portfile:
```diff
@@ -0,0 +1,76 @@
+
+set kf5.branch      5.47
+
+if {${subport} eq "${name}-devel"} {
```
There is no point in creating a -devel port for KI18n. It's fine (for a -hopefully- temporary KI18n port) to adapt the ECM Portfile, but I suggest you clean it up.

RJVB commented on kde/ki18n/Portfile:
```diff
@@ -0,0 +1,76 @@
+platforms           darwin
+homepage            https://projects.kde.org/projects/kdesupport/ki18n
+
+qt5.depends_component \
```
KI18n v5.42.0 also depended on QtDeclarative. Please double-check; a priori QtScript is obsolete and code should migrate to QML so it seems unlikely that the dependency has been dropped.

yan12125 replied:
Looks like QtDeclarative/QML is used in tests only - https://github.com/KDE/ki18n/search?utf8=%E2%9C%93&q=qml&type=, and there's currently no `test` defined in this port. (Of course adding test definitions would be nice) On the other hand, QtScript is recommended by KDE developers:
```
option(BUILD_WITH_QTSCRIPT "Build with support for scripted translations (recommended)" ON)
```
(https://github.com/KDE/ki18n/blob/f935c698b5ad1fb55fbd093ea1719d7a38f7f564/CMakeLists.txt#L34)

chrstphrchvz replied:
The dependencies I put were naïvely based on checking what other package managers use/getting through the configure phase successfully (rather than knowing what's in ki18n), which is where I saw QtScript but no mention of QtDeclarative.
<span><details><summary>Configure phase output:</summary>
<pre>
--->  Configuring kf5-ki18n
        (using ccache)
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build" && /opt/local/bin/cmake -G "CodeBlocks - Unix Makefiles" -DCMAKE_BUILD_TYPE=MacPorts -DCMAKE_INSTALL_PREFIX="/opt/local" -DCMAKE_INSTALL_NAME_DIR="/opt/local/lib" -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/usr" -DCMAKE_C_COMPILER_LAUNCHER=/opt/local/bin/ccache -DCMAKE_CXX_COMPILER_LAUNCHER=/opt/local/bin/ccache -DCMAKE_C_COMPILER="$CC" -DCMAKE_CXX_COMPILER="$CXX" -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_POLICY_DEFAULT_CMP0060=NEW -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_MODULE_PATH="/opt/local/share/cmake/Modules" -DCMAKE_PREFIX_PATH="/opt/local/share/cmake/Modules" -DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=ON -DCMAKE_INSTALL_RPATH="/opt/local/lib" -Wno-dev -DECM_MKSPECS_INSTALL_DIR=/opt/local/libexec/qt5/mkspecs/modules -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.13" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/ki18n-5.47.0 
-- The C compiler identification is AppleClang 9.1.0.9020039
-- The CXX compiler identification is AppleClang 9.1.0.9020039
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- 

-- Looking for __GLIBC__
-- Looking for __GLIBC__ - not found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Could not set up the appstream test. appstreamcli is missing.
CMake Warning at /opt/local/share/ECM/kde-modules/KDECMakeSettings.cmake:270 (message):
  Searching for X11 and related technologies is disabled on Apple systems.
  Set APPLE_FORCE_X11 to ON to change this behaviour.  Set
  APPLE_SUPPRESS_X11_WARNING to ON to hide this warning.
Call Stack (most recent call first):
  CMakeLists.txt:41 (include)


-- Looking for dngettext
-- Looking for dngettext - not found
-- libintl is a separate library.
-- Found LibIntl: /opt/local/include  
-- Found PythonInterp: /usr/bin/python (found version "2.7.10") 
-- Found Gettext: /opt/local/bin/msgmerge (found version "0.19.8.1") 
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- The sip executable must be available to use PythonModuleGeneration.
-- The following REQUIRED packages have been found:

 * ECM (required version >= 5.47.0), Extra CMake Modules., <https://projects.kde.org/projects/kdesupport/extra-cmake-modules>
 * Qt5Script (required version >= 5.8.0)
 * Qt5Core
 * LibIntl, <http://gnuwin32.sourceforge.net/packages/libintl.htm>
   Needed for building KI18n unless glibc is the system libc implementation
 * PythonInterp
 * Gettext
 * Qt5Test
 * Qt5Network (required version >= 5.10.1)
 * Qt5Qml
 * Qt5Concurrent
 * Qt5 (required version >= 5.8.0)

-- The following features have been disabled:

 * QCH, API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)

-- The following OPTIONAL packages have not been found:

 * PythonModuleGeneration

-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build
</pre>
</details><br /></span>

If I add `test.run yes` to the Portfile I can get these test results:
```
--->  Testing kf5-ki18n
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build"&& /usr/bin/make test
Running tests...
/opt/local/bin/ctest --force-new-ctest-process
Test project /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build
    Start 1: ki18n-declarativetest
1/5 Test #1: ki18n-declarativetest ............   Passed    0.10 sec
    Start 2: ki18n-klocalizedstringtest
2/5 Test #2: ki18n-klocalizedstringtest .......   Passed    0.08 sec
    Start 3: ki18n-ktranscripttest
3/5 Test #3: ki18n-ktranscripttest ............***Failed    0.06 sec
    Start 4: ki18n-ktranscriptcleantest
4/5 Test #4: ki18n-ktranscriptcleantest .......   Passed    0.06 sec
    Start 5: ki18n_install
5/5 Test #5: ki18n_install ....................   Passed    5.55 sec

80% tests passed, 1 tests failed out of 5

Total Test time (real) =   5.86 sec

The following tests FAILED:
          3 - ki18n-ktranscripttest (Failed)
Errors while running CTest
make: *** [test] Error 8
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build" && /usr/bin/make test
Exit code: 2
```

chrstphrchvz commented:
@RJVB Thanks for the quick feedback. You have a point with whether this is worthwhile. My only interest in Falkon/QupZilla is really for it being known as a "lightweight" browser, good for running on my older Linux machines or Raspberry Pi. Since my Macs don't have a problem with heavier browsers, I'm not personally interested in running Falkon on a Mac.

Also, I should've come across [this announcement](https://www.falkon.org/2018/05/08/falkon-301-released/) earlier:
> State of macOS port is currently in bad shape, so there won’t be any macOS builds until the situation changes.

Should this update be put "on hold" until (1) the upstream Falkon-on-macOS situation improves and/or (2) someone speaks up about wanting a Falkon port?

RJVB commented:
> My only interest in Falkon/QupZilla is really for it being known as a "lightweight" browser, good for running on my older Linux machines or Raspberry Pi.

Any browser using QtWebEngine doesn't really qualify as lightweight in my book. For that, I'd have a look at Otter-Browser using the rebooted QtWebKit .

> Since my Macs don't have a problem with heavier browsers, I'm not personally interested in running Falkon on a Mac.

Then I'd strongly suggest to spent your time on a more (personally) worthwhile project!

> Should this update be put "on hold" until (1) the upstream Falkon-on-macOS situation improves and/or (2) someone speaks up about wanting a Falkon port?

Probably both, or (3) when the port maintainer himself decides to update the port.

"currently in bad shape", not exactly specific, can mean anything, but any project pulling in KF5 frameworks without considering the cross-platform implications that brings along will run into issues sooner or later. With KI18n it is quite likely that the application will look in the wrong location for the translation files, when using stock Qt5.

RJVB commented:
>Looks like QtDeclarative/QML is used in tests only - https://github.com/KDE/ki18n/search?utf8=%E2%9C%93&q=qml&type=, and there's currently no `test` defined in this port.

That's possible, I haven't checked why the dependency is needed (as it's installed anyway because required by other frameworks).

>(Of course adding test definitions would be nice)

This may or may not  work correctly with an unpatched Qt5, so I wouldn't bother. They're mostly intended for developers working *on* KI18n anyway.

>On the other hand, QtScript is recommended by KDE developers:
>```
>option(BUILD_WITH_QTSCRIPT "Build with support for scripted translations (recommended)" ON)

Sure, but that's an old recommendation that probably predates the QtScript obsoletion. At some point this will have to be refactored to use QML, or some other scripting language. (KTextEditor already migrated several releases ago.)

yan12125 commented:
> Sure, but that's an old recommendation that probably predates the QtScript obsoletion. At some point this will have to be refactored to use QML, or some other scripting language. (KTextEditor already migrated several releases ago.)

I checked a few files in KI18N and it looks like QtScript is necessary only if a program uses KI18N to translate strings in scrtips, IIUC. I believe KI18N are primarily used by KDE programs and it's unlikely they are still using QtScript. If that's indeed the case, it's fine to drop the QtScript dependency from KI18N.

RJVB commented:
For a temporary port and/or for purposes of building Falkon, port:kf5-ki18n could probably indeed be built with -DBUILD_WITH_QTSCRIPT=OFF. I wouldn't do that for a generic build.

RJVB commented:
Now, about port:kf5-ki18n ... we could try to set up a PR for my KF5-Frameworks metaport once I have upgraded it to 5.47.0 somewhere next week (I hope). "We" would be me + @chrstphrchvz and/or @yan12125.

That metaport has subports for all relevant KF5 Frameworks but we could start with just the Tier1 frameworks (for the PR; commenting out the rest) and then take it from there.

I've tried to ensure that everything builds against the stock Qt5 port (and runs, as far as that's possible) but have never been able to test this so that would be the 1st thing to check. I hope only small changes will be required, if any..

RJVB commented:
On Friday June 15 2018 22:21:13 Chih-Hsuan Yen wrote:
>> Now, about port:kf5-ki18n ... we could try to set up a PR for my KF5-Frameworks metaport once I have upgraded it to 5.47.0 somewhere next week (I hope). "We" would be me + @chrstphrchvz and/or @yan12125.
>
>Thank you very much for the efforts! FWIW, I have some patches against MacStrop that fixes building with stock Qt 5.10 at https://github.com/yan12125/macstrop/commits/master. During the last time I checked, Okular runs fine after manually creating symlinks for XDG paths.
>
>

Thanks, I'll merge your changes to the frameworks (probably introducing at least 1 conflict for your repo, I'm afraid). I'll hold off on the Okular bump because I *really* need to upgrade all KF5 applications...

I merged the osx-integration fixes but for its Portfile I'll need to come up with a reasonable way to select the correct branch because the 5.10 version doesn't work very nicely against Qt 5.9 . Maybe the 5.10 version will work with Qt 5.11 too though?

Indeed, creating symlinks will work at least for simpler applications, but is a tricky business to get right programmatically from the Portfile.

chrstphrchvz commented:
Pushed new commits rebased from cb73e44. In particular for 366def2 I corrected the port name to `kf5-ki18n`, removed its `-devel` subport, and enabled tests for it.

@RJVB @yan12125 Please let me know if I should put/add either of you as maintainer for kf5-ki18n. I don't mean to dump more work onto others, but both of you seem far more knowledgable about these packages than I am.

RJVB commented:
>The dependencies I put were naïvely based on checking what other package managers use/getting through the configure phase successfully (rather than knowing what's in ki18n), which is where I saw QtScript but no mention of QtDeclarative.

If you look at the CMake output:
```
 * Qt5Qml
```

That's provided by QtDeclarative.

BTW, this is part of the reason why I implemented `cmake.save_configure_cmd` with the `"log too"` option: this saves the cmake invocation and output in separate files in the port's work directory in addition to the *volatile* output embedded in the logfile.

> @RJVB @yan12125 Please let me know if I should put/add either of you as maintainer for kf5-ki18n. I don't 

It would definitely make sense to put me on it, being the maintainer of the (only) existing KF5 ports tree.
Does yan12125 have commit access, and/or does he intend to step up as a regular co-maintainer? If so he is more than welcome (you too), but with the caveat emptor that I've invested too much time in (and rely too much on) my KF5 ports to give out/up the reins ;)


But isn't all this a bit dependent on whether or not you're indeed going to push for a Falkon upgrade? I'd advise against volunteering to become maintainer of a port that you're not interested in yourself. (In this case you might probably take the fact that the current QupZilla maintainer wasn't aware of the name-change as a telltale sign that there isn't that much interest to this browser - on Mac. Is it really more lightweight than Safari, for instance?)

Let me repeat: if you're looking for a challenge, let's start with getting my `port:KF5-Frameworks` with the 2 KF5 PortGroup files ready for a PR, at least the tier1 ports. With ready I mean tested against the `port:qt5` family of Qt5 components and if possible on more than a single OS version. I understand Yan has already worked on this so there may be not much more left to do than to decide whether to out-comment or strip the other tier frameworks.

This would give a port:kf5-ki18n without need for introducing a temporary port, and with a bit of luck it'll  get KF5 some traction. I have updated the Tier1 and Tier2 ports to 5.47.0 this weekend, so there's not even a need to wait for that anymore.

R.

yan12125 commented:
I looked into Falkon sources, and looks like KI18N is used for an example plugin `hellopython` only. There are three possible approaches to get updated Falkon -

1. Introduce a standalone ki18n port and make Falkon depend on it
2. Wait for RJVB's KF5 port
3. Patch Falkon so that it doesn't depend on ki18n. Looks like this is as simple as removing the `po/` directory from the sources.

I'm -1 on option 1 as it introduces duplicated work, which is not a good scene. I have no preference between 2 and 3.

> Does yan12125 have commit access, and/or does he intend to step up as a regular co-maintainer? If so he is more than welcome (you too), but with the caveat emptor that I've invested too much time in (and rely too much on) my KF5 ports to give out/up the reins ;)

Yes I have commit access and I'm willing to maintain KDE-related ports. I have some softwares that I want to put into MacPorts and they depend on KDE stuffs. Anyway this is not an issue if we go option 2 or 3.

pmetzger commented:
So where are we on this?

yan12125 commented:
RJVB's extra-cmake-modules is upgraded! @RJVB: could you submit the updated extra-cmake-modules, and then this PR can be rebased against that.

chrstphrchvz commented:
I may have figured out the cause of the CMake error; I have [submitted it upstream](https://phabricator.kde.org/D13736) and can include a patch in MP for now.

However, the build only gets about 76% of the way through before an error:
```
In file included from /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_falkon/falkon/work/build/src/lib/FalkonPrivate_autogen/mocs_compilation.cpp:54:
/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_falkon/falkon/work/build/src/lib/FalkonPrivate_autogen/CB6E7VNMGS/moc_mactoolbutton.cpp:81:25: error: call to non-static member function without an object argument
    return QToolButton::qt_metacast(_clname);
           ~~~~~~~~~~~~~^~~~~~~~~~~
/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_falkon/falkon/work/build/src/lib/FalkonPrivate_autogen/CB6E7VNMGS/moc_mactoolbutton.cpp:86:24: error: call to non-static member function without an object argument
    _id = QToolButton::qt_metacall(_c, _id, _a);
          ~~~~~~~~~~~~~^~~~~~~~~~~
```
The same error happens in the master branch.

chrstphrchvz commented:
I reported this build error upstream: https://bugs.kde.org/show_bug.cgi?id=395926

Here's a complete build log, which I neglected to provide earlier: https://gist.github.com/chrstphrchvz/cc5979a7ab5f807fbad47eba3fb0a9ce

chrstphrchvz commented:
I already got a reply from falkon project closing that issue:
> --- Comment #1 from David Rosca <nowrep@gmail.com> ---
>Mac is not supported platform, due to issues with compilation and at runtime.
>Please post patches instead of reporting bugs.

RJVB commented:
I think that kind of answers the question whether it's worth the trouble of supporting this project in MacPorts...

chrstphrchvz commented:
So for now, do we keep qupzilla as-is, or schedule it to be removed in a year (if not sooner)?

yan12125 commented:
IMO old browsers shouldn't be kept for a long time as it will be hit by security issues. That's a reason I put efforts into updating webkit2-gtk.

RJVB commented:
As I've mentioned before, QupZilla or falkon are only as secure as the browser engine they depend on (QtWebEngine, QWE).

As long as Qupzilla works with the latest QWE it should be safe enough, in any case for users with a minimum of common sense (which I'd expect from MacPorts users as the whole project isn't exactly for complete dummies ;) ).

yan12125 commented:
> As long as Qupzilla works with the latest QWE it should be safe enough

Yep. That would be a reasonable criteria for how long MacPorts should keep QupZilla.

pmetzger commented:
Okay, so where are we on this PR? What's the next step on it and who is taking that step?

chrstphrchvz commented:
@pmetzger Because upstream isn't supporting macOS at this time, and I don't have the knowledge/resources to get it working to a point where it might be supported, I'm thinking the QupZilla&rarr;Falkon change is going to be abandoned, with QupZilla being kept as-is for the time being as long as its dependencies are maintained/remain compatible. 

Regarding the current build error: even though I understand the error in general, I don't know how Qt is generating the mocs_*.cpp files and how I might fix it.

@RJVB and @yan12125 are free to reuse anything, especially the first two commits (ECM and ki18n; no credit to me is necessary for those). I'm fine with closing this PR and leaving the branch intact in case either (1) someone more ambitious wants to reuse this later and/or (2) Falkon is supported on macOS upstream.

RJVB commented:
I'm thinking that yan12125 has tested my KF5 frameworks against the stock port:qt5, so I'm thinking I, or rather we, could make a PR out of the Tier1 part of my KF5 frameworks meta-port and the 2 PortGroup files that go with it.

I say we, because for this kind of endeavour I'd really prefer to have a co-maintainer willing to do the PR stuff (pun intended) while I keep my energy and sanity focussed on the actual development stuff.

And it would be convenient for me to postpone this until after summer (holidays & heat-waves)...

R.

chrstphrchvz commented:
@pmetzger since the goal of this PR isn't attainable for now, I think it should be closed; I'll go ahead and do so.

Long story short, this was not the trivial update I was hoping for (biting off more than I can chew, so to speak). Though this does give me a better understanding of the MP ecosystem. Thanks @RJVB and @yan12125 for your guidance.

yan12125 commented:
@RJVB: feel free to ping me for any KDE-related stuff :)
